src/model_signing/model.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2024 The Sigstore Authors
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pathlib
+from typing import Callable, TypeAlias
+
+from model_signing.manifest import manifest
+from model_signing.serialization import serialization
+from model_signing.signature import verifying
+from model_signing.signing import signing
+
+
+PayloadGeneratorFunc: TypeAlias = Callable[
+    [manifest.Manifest], signing.SigningPayload
+]
+
+
+def sign(
+    model_path: pathlib.Path,
+    signer: signing.Signer,
+    payload_generator: PayloadGeneratorFunc,
+    serializer: serialization.Serializer,
+    ignore_paths: list[pathlib.Path] | None = None,
```

Comment:
What is the difference between `None` and `[]`

Reply:
But you can have `ignore_paths: list[pathlib.Path] = []`

Reply:
Just FYI: `ignore_paths: list[pathlib.Path] = []` that does not work with the current linter settings